src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs

```diff
@@ -575,6 +577,34 @@ internal static void ThrowFormatIndexOutOfRange()
             throw new FormatException(SR.Format_IndexOutOfRange);
         }
 
+        [DoesNotReturn]
+        internal static void ThrowAmbiguousMatchException(MemberInfo memberInfo)
+        {
+           throw GetAmbiguousMatchException(memberInfo);
+        }
+
+        internal static AmbiguousMatchException GetAmbiguousMatchException(MemberInfo memberInfo)
+        {
+            var declaringType = memberInfo.DeclaringType;
+            return (declaringType is not null)
+                ? new AmbiguousMatchException(SR.Format(SR.Arg_AmbiguousMatchException_Detailed, declaringType.ToString(), memberInfo.ToString()))
```

Comment:
Using the example of being passed the `MethodInfo` of  the property`System.String.Length`....

So if we **do have** a non-null `.DeclaringType`, we can use that to format up with more detail (yielding for example "System.String" for the first part.

If we **do not**, then we just emit the `memberInfo.ToString()` (which yields for example "Int32 Length")

Reply:
Not sure if adding the `DeclaringType` info that useful ...